### PR TITLE
mekf: flip imu data, fix update step

### PR DESCRIPTION
**Pull request by oyqmatt**

### CODE REVIEW

File: src/estimation/src/mekf6.cpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ Mekf6::GetQMatrix(double dt) {
   Q.block<3, 3>(6, 12) = -m_sigma_beta_f * dt * dt * dt / 6.0f;
 
   Q.block<3, 3>(9, 0) = -m_sigma_beta_omega * dt * dt / 2.0f;
-  Q.block<3, 3>(9, 9) = m_sigma_beta_omega * dt * dt / 2.0f;
+  Q.block<3, 3>(9, 9) = m_sigma_beta_omega * dt;
```
Comment (rxdu) [inline src/estimation/src/mekf6.cpp:68]: we need to verify if it's the error in the paper or in the python reference code, but the difference should be minor

File: src/estimation/src/mekf6.cpp [hunk 2]
```diff
@@ -152,5 +151,7 @@ void Mekf6::Update(const ControlInput &gyro_tilde,
   q_hat_ =
       q_hat_ * Eigen::Quaterniond(1, x_(0) / 2.0f, x_(1) / 2.0f, x_(2) / 2.0f);
   q_hat_.normalize();
+
+  x_.segment<3>(0) = Eigen::Vector3d::Zero();
```
Comment (rxdu) [inline src/estimation/src/mekf6.cpp:155]: why reset the quaternion errors to 0?

File: src/estimation/src/mekf6.cpp [hunk 3]
```diff
@@ -87,8 +87,8 @@ void Mekf6::Update(const ControlInput &gyro_tilde,
 
   // predict state with control input
   q_hat_ = Eigen::Quaterniond(
-      q_hat_.coeffs() +
-      0.5 * dt * Eigen::Quaterniond(0, gyro(0), gyro(1), gyro(2)).coeffs());
+      q_hat_.coeffs() + (q_hat_ *Eigen::Quaterniond(
```
Comment (rxdu) [inline src/estimation/src/mekf6.cpp:90]: I indeed missed the term q_hat_
```
Eigen::Quaterniond(q_hat_.coeffs() + 0.5 * dt * (q_hat_ *Eigen::Quaterniond(0, gyro(0), gyro(1), gyro(2)).coeffs()).coeffs());
```

File: src/estimation/src/mekf6.cpp (hunk 1 above)
Comment (oyqmatt) [reply src/estimation/src/mekf6.cpp:68]: This one is following the python code. The paper by James M Maley doesnt provide Q for 6 axis, only 9 axis, so have to derive. Is your version from another paper?

**Pull request closed without merging**